### PR TITLE
[MM-34325] Migrate 'components/admin_console/permission_schemes_settings' and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by rubenmeza**
#### Summary

This code migrate 'components/admin_console/permission_schemes_settings' components to typescript

#### Ticket Link

Github Issue: mattermost/mattermost-server#17250
JIRA: https://mattermost.atlassian.net/browse/MM-34325

#### Release Note

```release-note
- Refactor component to typescript, no changes in UI
```

### CODE REVIEW

**Comment by rubenmeza** [comment]
Hello @devinbinnie Just a question, what would be the correct type for isDisabled prop?, since, in the component is defined as a boolean but PermissionsSchemeSummary requires a function

**Comment by devinbinnie** [comment]
> Hello @devinbinnie Just a question, what would be the correct type for isDisabled prop?, since, in the component is defined as a boolean but PermissionsSchemeSummary requires a function

@rubenmeza Looks like `permissions_scheme_summary` was migrated incorrectly. We can change the prop type in that component to `boolean` since that makes more sense. Thanks for catching that :)

**Comment by rubenmeza** [comment]
Hello @devinbinnie this is ready for review 🚀

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks @rubenmeza!

**Review by calebroseland — CHANGES_REQUESTED** [review]
Hey @rubenmeza, sorry for the delay, and thanks for catching/fixing the `isDisabled` discrepancy!

Overall this looks good, but I made a patch with some suggestions. The main parts seemed to be `schemes: []`, and an overlap with `history?: []` and `RouteComponentProps`. (I was seeing runtime proptype warnings for these.)
[9331.patch.zip](https://github.com/mattermost/mattermost-webapp/files/7542250/9331.patch.zip)

I'd also like to suggest migrating the index if that isn't too much to ask.

**Comment by rubenmeza** [comment]
Hello @calebroseland no problem I applied the patch and also migrate the index file, can you please review it again

File: components/admin_console/permission_schemes_settings/index.tsx [hunk 1]
```diff
@@ -21,9 +25,9 @@ function mapStateToProps(state) {
     };
 }
 
-function mapDispatchToProps(dispatch) {
+function mapDispatchToProps(dispatch: Dispatch<GenericAction>) {
     return {
-        actions: bindActionCreators({
+        actions: bindActionCreators<ActionCreatorsMapObject<ActionFunc| GenericAction>, Props['actions']>({
```
Comment (calebroseland) [inline components/admin_console/permission_schemes_settings/index.tsx:30]: just for consistent spaces, but not a big deal 0/5
```suggestion
        actions: bindActionCreators<ActionCreatorsMapObject<ActionFunc | GenericAction>, Props['actions']>({
```

**Review by calebroseland — APPROVED** [review]
LGTM, thanks @rubenmeza!

File: components/admin_console/permission_schemes_settings/index.tsx (hunk 1 above)
Comment (rubenmeza) [reply components/admin_console/permission_schemes_settings/index.tsx]: fixed on 6f2facaf2fcb999741899fa9e461cf4a4cd5bcf7

**Review by calebroseland — APPROVED** [review]

File: components/admin_console/permission_schemes_settings/permission_schemes_settings.tsx [hunk 2]
```diff
@@ -16,27 +17,37 @@ import LoadingWrapper from 'components/widgets/loading/loading_wrapper';
...
-        isDisabled: PropTypes.bool,
+export type Props = {
+    schemes: SchemesState['schemes'];
+    jobsAreEnabled?: boolean;
+    clusterIsEnabled?: boolean;
+    license: {
+        CustomPermissionsSchemes: string;
+    };
+    actions: {
+        loadSchemes: (a: string, b: number, c: number) => Promise<ActionResult>;
```
Comment (devinbinnie) [inline components/admin_console/permission_schemes_settings/permission_schemes_settings.tsx:37]: Can we have the arguments match the original method signature?
```suggestion
        loadSchemes: (scope: SchemeScope, page: number, perPage: number) => Promise<ActionResult>;
```

**Review by devinbinnie — CHANGES_REQUESTED** [review]
Thanks @rubenmeza! Just one comment from my end.

**Review by calebroseland — APPROVED** [review]

**Review by devinbinnie — APPROVED** [review]
Thanks @rubenmeza, this is looking good now :)

**Comment by devinbinnie** [comment]
@furqanmlk Gentle ping to review :)

**Review by furqanmlk — APPROVED** [review]
E2E Passed https://automation-dashboard.vercel.app/cycles/8aca8f0f-b0a9-475d-a54b-7f41d827ebce
And Smoke test passed ✅

**Pull request merged**